compiler/rustc_next_trait_solver/src/solve/eval_ctxt/mod.rs

```diff
@@ -1001,12 +1001,8 @@ where
     // Try to evaluate a const, or return `None` if the const is too generic.
     // This doesn't mean the const isn't evaluatable, though, and should be treated
     // as an ambiguity rather than no-solution.
-    pub(super) fn try_const_eval_resolve(
-        &self,
-        param_env: I::ParamEnv,
-        unevaluated: ty::UnevaluatedConst<I>,
-    ) -> Option<I::Const> {
-        self.delegate.try_const_eval_resolve(param_env, unevaluated)
+    pub(super) fn evaluate_const(&self, param_env: I::ParamEnv, ct: I::Const) -> Option<I::Const> {
```

Comment:
What's the benefit of taking a `I::Const` here rather than the `UnevaluatedConst` directly? 

I think that making the input type *more specific* would make it less confusing that we only call today for uv consts